### PR TITLE
Fix ada support

### LINKED ISSUE #302: Fix ADA support
Our ADA support right now is broken. We need someone to look at what we currently have, research how ADA addresses are support to work, and then submit a PR for a fix.

Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space. If your PR expands the file size of the library too much, we won't accept it.

We will likely be reviewing PRs on a regular basis. No need to ping us to check your PR, we will definitely get to it.

Thanks for your help!

### PR DESCRIPTION
**Pull request by Echolon166**
## Issue number

Closes #302

## Description

Fixed Shelley address implementation, added missing Byron address implementation.

## Reference to the specification

- [CIP-0008](https://github.com/cardano-foundation/CIPs/blob/9e410b0ce3d18b0e3f2d0ba3fc8ad431757c2aa2/CIP-0008/CIP-0008.md)
- [Byron address format](https://cardanoupdates.com/docs/767c8e3d-c0be-4789-bdb0-bda4c3f5abb6)

## Reference to the test address.

- [CIP-0019](https://github.com/cardano-foundation/CIPs/blob/85cf8778fa18dbf609224c890902bc38a51544fc/CIP-0019/CIP-0019.md#:~:text=type-00,addr1w8phkx6acpnf78fuvxn0mkew3l0fd058hzquvz7w36x4gtcyjy7wx)

## List of features added/changed

- Add cardanoChain

## How much has the filesize increased?
Increased +40k due to crc library dependency, ported cbor since their latest release is missing on the npm, which has critical changes.

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.
- [x] I have specified correct coinTypes specified at [Slip 44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md)
- [x] I have provided the reference link to the specification I implemented.
- [x] I have provided enough explanation about how I provided the test address

### CODE REVIEW

File: src/cbor/cbor.ts [hunk 1]
```diff
@@ -0,0 +1,540 @@
+/* tslint:disable:max-classes-per-file */
```
Comment (Arachnid) [inline src/cbor/cbor.ts:1]: Can you please include source and licensing information here?

Comment (Echolon166) [reply src/cbor/cbor.ts:3]: Done. 
A bit of a correction about the reason of the port: Their latest release is missing on the npm, and adding the github repo as dependency forces different utilities such as gulp.

Comment (Arachnid) [reply src/cbor/cbor.ts:3]: Are there not any alternative CBOR libraries that could be used from NPM?

I'm a bit concerned about the amount this increases the package size by, for just one chain.

Comment (Echolon166) [reply src/cbor/cbor.ts:3]: Tried different crc implementations and replaced crc dependency with best/smallest one.

Updated filesize increase:
378286 -> 387385 (~9k)

**Pull request merged**